internal/dag/gatewayapi_processor.go

```diff
@@ -71,6 +71,14 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 		var matchingRoutes []*gatewayapi_v1alpha1.HTTPRoute
 		var listenerSecret *Secret
 
+		// Validate the Kind on the selector is a supported type.
+		if listener.Protocol != gatewayapi_v1alpha1.HTTPProtocolType &&
+			listener.Protocol != gatewayapi_v1alpha1.HTTPSProtocolType &&
+			listener.Protocol != gatewayapi_v1alpha1.TLSProtocolType {
+			p.Errorf("Listener.Protocol %q is not supported.", listener.Protocol)
+			continue
+		}
+
```

Comment:
based on the API docs Sunjay linked, it sounds like we should validate that if the protocol is HTTPS/TLS, that `listener.TLS` is non-nil (but if the protocol is something else, we just ignore `listener.TLS`).

Reply:
Ok I'll do that and open an issue upstream. This protocol specification doesn't seem clear to me. If you specify on an HTTP protocol, just ignore, but no error if you don't specify on an HTTPS protocol? Right now this just uses if it's there and ignores if not.

Reply:
The other unclear bit is if we specify TLS on the HTTPRoute (which isn't yet added to Contour), what happens then? 

// ref: https://gateway-api.sigs.k8s.io/references/spec/#networking.x-k8s.io/v1alpha1.RouteTLSConfig

Reply:
Sounds like in that case `listener.TLS` still needs to be specified with a mode of `Passthrough`